### ISSUE TITLE
RELEASE.TXT is not in my NDK

### ISSUE THREAD

**Issue by renaudfavier** [realm/realm-java#3284]
Added by @kneth: When we upgrade to NDK >= r11, we will have to find another method to validate NDK version.

#### Goal

I'm just trying to make the example projects work. 
I've cloned the entire example repo so I executed `./graddlew assemble`, He told me that i needed the NDK and its environment variable. So I downloaded [this ndk](http://dl.google.com/android/repository/android-ndk-r12b-linux-x86_64.zip) and set the NDK_HOME to the correct location.
=> new error : *"The path provided in the NDK_HOME environment variable does not seem to be an Android NDK."*

So I looked a bit in your code and found :
``` 
if (!file("${ndkDir}/RELEASE.TXT").file) {
        throw new GradleException('The path provided in the NDK_HOME environment variable does not seem to be an Android NDK.')
}
```
(here : https://github.com/realm/realm-java/blob/master/realm/realm-jni/build.gradle line 130)

well my ndk does not have this file indeed, I have the feeling that it was there on previous versions of NDK and that's what is the problem so I posted this issue.

Thanks,

```

#### Version of Realm and tooling
Realm version(s): just the examples

Android Studio version: 2.1

Which Android version and device: -

**Comment by Zhuinden** [comment]
I'm not a Realm person so don't be surprised by my statement, but you should try putting that TXT there just so it would find the file and proceed.

**Comment by beeender** [comment]
currently compiling Realm requires ndk r10e , so please just install that version.

The later NDK seems not have the `RELEASE.txt`, we need to figure out the other way to check the ndk version.

**Comment by kneth** [comment]
See also https://github.com/android-ndk/ndk/issues/40

**Comment by renaudfavier** [comment]
Ok I see, thanks for answering this fast.

**Comment by kneth** [comment]
@renaudfavier You can argue that you have filed a bug report before it happens :-) I mean, when we upgrade to a newer NDK, this will happen. And we will have to deal with it.

**Comment by emanuelez** [comment]
Just for future reference: in newer NDKs the version is contained in the source.properties file, but it appears in another format. This is r12 for example:

```
$ cat source.properties
Pkg.Desc = Android NDK
Pkg.Revision = 12.0.2931149
```

**Comment by beeender** [comment]
i am not quite sure if `2931149` changes, will lto linking fail :(

**Comment by renaudfavier** [comment]
Aha yes one can see it like that !

I take the occasion to ask you about the immediate next problem I had with the same `/gradlew assemble` : 
```
A problem occurred evaluating project ':realm-library'.
> failed to find Build Tools revision 24.0.0
```
My ANDROID_HOME  points well to my Sdk which has "24.0.0-preview" and "24.0.1" but no 24.0.0

**Comment by Zhuinden** [comment]
@renaudfavier you'll need to install Build Tools 24 from the SDK Manager

**Comment by renaudfavier** [comment]
@Zhuinden Yes what I was trying to say is that I don't think 24.0.0 exists, if i `./tools/android list sdk -a | grep Build-tools` in my Sdk here is the output : 

```
   4- Android SDK Build-tools, revision 24.0.1
   5- Android SDK Build-tools, revision 24
   6- Android SDK Build-tools, revision 23.0.3

```
and revision 24 points to 24.0.0-preview
all my builds tools are up to date, sdk manager points to them.

I think this might be the problem because this is the first time I see a "-preview"

**Comment by zaki50** [comment]
@renaudfavier Has this issue been solved?

**Comment by zaki50** [comment]
fixed by #3316
